packages/bullmq/package.json

```diff
@@ -30,14 +30,14 @@
     "@types/jest": "29.5.11",
     "@types/node": "20.11.0",
     "@types/reflect-metadata": "0.1.0",
-    "bullmq": "4.13.0",
+    "bullmq": "5.1.1",
     "reflect-metadata": "0.2.1",
     "rxjs": "7.8.1"
   },
   "peerDependencies": {
     "@nestjs/common": "^8.0.0 || ^9.0.0 || ^10.0.0",
     "@nestjs/core": "^8.0.0 || ^9.0.0 || ^10.0.0",
-    "bullmq": "^3.0.0 || ^4.0.0"
+    "bullmq": "^5.0.0"
```

Comment:
Is it really necessary to drop support for `bullmq@3` and `bullmq@4` because of these changes?

Reply:
this one is a concern, that may not work for 3 and 4

![image](https://github.com/nestjs/bull/assets/27850115/7f101770-9d25-4bfd-ba3a-f4b5bfa3a5c2)

Reply:
what happens if we pass the `sharedConnection` attribute to the bullmq@5 worker?

Reply:
I think nothing, it's not present there anymore and we don't have it in queueOpts too, because we do use their type

Reply:
@kamilmysliwiec do you want to go this path? I can custom type of the queue opts where we always will have sharedConnection, so it will be just confusing for v5 users, if they pass it it won't be used, that shouldn't be a big deal as for me

Reply:
I'm OK with whatever helps us _not to introduce a breaking change_ so we can support bullmq sooner

Reply:
@kamilmysliwiec ready, I tested it by installing locally v3, v4, v5 and running build and test

Reply:
just FYI, I meant that PR is ready from my side, so if you can review and merge we will have a bull5 support